homeassistant/components/calendar/todoist.py

```diff
@@ -0,0 +1,595 @@
+        # complete the task.
+        # Generally speaking, that means right now.
+        task['start'] = dt.utcnow()
+        if data['due_date_utc'] is not None:
+            due_date = data['due_date_utc']
+
+            # Due dates are represented in RFC3339 format, in UTC.
+            # Home Assistant exclusively uses UTC, so it'll
+            # handle the conversion.
+            time_format = '%a %d %b %Y %H:%M:%S %z'
+            # HASS' built-in parse time function doesn't like
+            # Todoist's time format; strptime has to be used.
+            task['end'] = datetime.strptime(due_date, time_format)
+
+            if self._latest_due_date is not None and (
+                    task['end'] > self._latest_due_date):
+                # This task is out of range of our due date;
+                # it shouldn't be counted.
+                return None
+
+            task['due_today'] = task['end'].date() == datetime.today().date()
+
+            # Special case: Task is overdue.
+            if task['end'] <= task['start']:
+                task['overdue'] = True
+                # Set end time to the current time plus 1 hour.
+                # We're pretty much guaranteed to update within that 1 hour,
+                # so it should be fine.
+                task['end'] = task['start'] + timedelta(hours=1)
+            else:
+                task['overdue'] = False
+        else:
+            # If we ask for everything due before a certain date, don't count
+            # things which have no due dates.
+            if self._latest_due_date is not None:
+                return None
+
+            # Define values for tasks without due dates
+            task['end'] = None
+            task['all_day'] = True
+            task['due_today'] = False
+            task['overdue'] = False
+
+        # Not tracked: id, comments, project_id order, indent, recurring.
+        return task
+
+    @staticmethod
+    def select_best_task(project_tasks):
+        """
+        Search through a list of events for the "best" event to select.
+
+        The "best" event is determined by the following criteria:
+          * A proposed event must not be completed
+          * A proposed event must have a end date (otherwise we go with
+            the event at index 0, selected above)
+          * A proposed event must be on the same day or earlier as our
+            current event
+          * If a proposed event is an earlier day than what we have so
+            far, select it
+          * If a proposed event is on the same day as our current event
+            and the proposed event has a higher priority than our current
+            event, select it
+          * If a proposed event is on the same day as our current event,
+            has the same priority as our current event, but is due earlier
+            in the day, select it
+        """
+        if len(project_tasks) > 0:
```

Comment:
You already make this check before you call this method. I suggest you assume you get a list and just iterate over that.